test/unit/transpiler/passes/scheduling/control_flow_test_case.py

```diff
@@ -0,0 +1,35 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022, 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Enhanced test case for control flow circuits."""
+
+from typing import Any, Optional
+
+from qiskit import QuantumCircuit
+from qiskit.test import QiskitTestCase
+from qiskit.test._canonical import canonicalize_control_flow
+
+
+class ControlFlowTestCase(QiskitTestCase):
```

Comment:
I don't think `QiskitTestCase` is used anywhere else in this repo. It does provide the benefit of turning deprecation warnings into errors, so perhaps that should be used more widely (obviously not this PR). 

But for now perhaps use `IBMTestCase` to keep things consistent.

Reply:
should `qiskit.test` be removed entirely as part of https://github.com/Qiskit/qiskit-ibm-runtime/pull/1292?

Reply:
Yes i'll use `IBMTestCase` in this PR and then remove all of qiskit.test in #1292